addons/l10n_sa/account.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import logging
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from operator import itemgetter
+import time
+
+import openerp
+from openerp import SUPERUSER_ID, api
+from openerp import tools
+from openerp.osv import fields, osv, expression
+from openerp.tools.translate import _
+from openerp.tools.float_utils import float_round as round
+from openerp.tools.safe_eval import safe_eval as eval
+
+import openerp.addons.decimal_precision as dp
+
+class account_journal(osv.osv):
+    _inherit = "account.journal"
+
+    _defaults = {
+        'entry_posted' : True,
+        'allow_date' : True,
+        }
+
+    def create_sequence(self, cr, uid, vals, context=None):
```

Comment:
I'm not creating journals in xml , it's created same way as original odoo journals hard coded and we are enforced to do it this way to override original names and codes and to get journals created for each company in multi-company.

based on this, if I want to set the sequence id on the journal I've no way to find the sequence defined in xml that's related to the journal.

Plz let me know if there's a way .